gtsam/geometry/Cal3.h

```diff
@@ -61,6 +61,133 @@ void calibrateJacobians(const Cal& calibration, const Point2& pn,
   }
 }
 
-//TODO(Varun) Make common base class for all calibration models.
+/**
+ * @brief Common base class for all calibration models.
+ * @addtogroup geometry
+ * \nosubgrouping
+ */
+class GTSAM_EXPORT Cal3 {
+ protected:
+  double fx_, fy_, s_, u0_, v0_;  ///< focal length, skew and principal point
+
+ public:
+  enum { dimension = 5 };
+  ///< shared pointer to calibration object
+  using shared_ptr = boost::shared_ptr<Cal3>;
+
+  /// @name Standard Constructors
+  /// @{
+
+  /// Create a default calibration that leaves coordinates unchanged
+  Cal3() : fx_(1), fy_(1), s_(0), u0_(0), v0_(0) {}
```

Comment:
refer to comment above. If defaults are like `double fx_ = 1;` then : `Cal3() = default;`.